module/users/helper.js

```diff
@@ -138,6 +138,33 @@ module.exports = class UsersHelper {
 				//   );
 				// result.programJoined = programJoinStatus.joinProgram;
 				// result.consentShared = programJoinStatus.consentShared;
+				let components = programData.components || []
+
+				if (components.length > 0) {
+					// Order solutions based on components order
+					let resultData = result.data
+
+					// Create a mapping of _id to order from components
+					const orderMap = new Map()
+					components.forEach((component) => {
+						orderMap.set(component._id.toString(), component.order)
+					})
+
+					// Sort resultData based on the order mapping
+					resultData = resultData
+						.map((item) => {
+							const order = orderMap.get(item._id.toString())
+							return { ...item, order: order !== undefined ? order : null }
+						})
+						.sort((a, b) => {
```

Comment:
@borkarsaish65 here also the variables are a,b

Reply:
done